### PR TITLE
🎉 bsdgames 🎉

### PR DESCRIPTION
**Pull request by saltedcoffii**
Classic text mode games from UNIX folklore.

https://github.com/skycocker/chromebrew/pull/5169#issuecomment-782291008

😜😜😜

### CODE REVIEW

**Comment by supechicken** [comment]
Wow, maybe this is the first game in our repo :P

File: packages/adventure.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    system "sed -i 's:-lncurses:-lncurses -ltinfo:g' configure" # Add libtinfo support
+    system "sed -i 's:-lcrypto:-lssl -lcrypto:g' configure" # Fix openssl
+    system "sed -i 's:#!/bin/sh:#!/bin/bash:' configure" # dash doesn't work
+    system "sed -i 's:read input:read -t 0.2 input:' configure" # Make script automatic
+  end
+  
+  def self.prebuild
+    puts 'Configuring'.lightblue
+    system "sed -i 's:-g -O2:-O2:g' configure"
+    system "sed -i 's:/usr/games:#{CREW_PREFIX}/bin:' wargames/wargames"
```
Comment (uberhacker) [inline packages/adventure.rb:57]: These sed commands could be moved to the patch section.

File: packages/adventure.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    system "echo bsd_games_cfg_build_dirs='\"#{@_bsdgame}\"' >> config.params"
+    system "echo bsd_games_cfg_docdir='#{CREW_PREFIX}/share/doc/bsdgames' >> config.params"
+    system "echo bsd_games_cfg_install_prefix='#{CREW_DEST_DIR}' >> config.params"
+    system "echo bsd_games_cfg_gamesdir='#{CREW_PREFIX}/bin' >> config.params"
+    system "echo bsd_games_cfg_sbindir='#{CREW_PREFIX}/sbin' >> config.params"
+    system "echo bsd_games_cfg_usrbindir='#{CREW_PREFIX}/bin' >> config.params"
+    system "echo bsd_games_cfg_libexecdir='#{CREW_PREFIX}/libexec' >> config.params"
+    system "echo bsd_games_cfg_man6dir='#{CREW_PREFIX}/share/man/man6' >> config.params"
+    system "echo bsd_games_cfg_man8dir='#{CREW_PREFIX}/share/man/man8' >> config.params"
+    system "echo bsd_games_cfg_man5dir='#{CREW_PREFIX}/share/man/man5' >> config.params"
```
Comment (uberhacker) [inline packages/adventure.rb:67]: You could use `#{CREW_MAN_PREFIX}` instead of `#{CREW_PREFIX}/share/man`.

File: packages/bsdgames.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+require 'package'
+
+class Bsdgames < Package
+  description 'A metapackage of classic text mode games from UNIX folklore.'
+  homepage 'https://www.polyomino.org.uk/computer/software/bsd-games/'
+  version '2.17'
+  compatibility 'all'
+  source_url 'file:///dev/null'
+  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
```
Comment (uberhacker) [inline packages/bsdgames.rb:9]: Instead of source_url and source_sha256, you could use `is_fake`.

**Comment by uberhacker** [comment]
@saltedcoffii has opened Pandora's box. 🙈 Do you think you could create a library to pass the game name and some parameters?  If possible, this would DRY up a lot of the duplicate patching in all packages.

**Comment by saltedcoffii** [comment]
> @saltedcoffii has opened Pandora's box. 🙈 Do you think you could create a library to pass the game name and some parameters? If possible, this would DRY up a lot of the duplicate patching in all packages.

I'm sure it is possble. It could just have all the traits of the package library or even be a subclass of package. The only variable in the repeating packages is `@_bsdgames` so that could definitely be incorporated into the library.

**Comment by saltedcoffii** [comment]
I do wanna write that library, however I don't know how to start. Could someone give me a basis or framework to work off of? Thanks ♥

**Comment by supechicken** [comment]
> I do wanna write that library, however I don't know how to start. Could someone give me a basis or framework to work off of? Thanks ♥

You can ask the user which game he wants using the 'gets' method (or use a `for` loop to install all games at once)

**Comment by saltedcoffii** [comment]
@uberhacker would you mind unmerging this so I can add the license '' thing to each package?

**Comment by uberhacker** [comment]
I think it would be easier to just add the 39 licenses.  They should all be close to the same, right?

**Comment by uberhacker** [comment]
Lesson learnt, hopefully.  Your working branch should always be in sync with master. :)

**Comment by saltedcoffii** [comment]
chromebrew needs to have a better way to test package files without first merging them with this master branch. I guess a ruby syntax checker isn't enough.

**Comment by uberhacker** [comment]
You are supposed to check and test locally before submitting a PR. :)

**Comment by satmandu** [comment]
> chromebrew needs to have a better way to test package files without first merging them with this master branch. I guess a ruby syntax checker isn't enough.

A good way to do this would be to change the default repo/branch for crew, since it uses git. This way you could check that everything works properly.

What do you think @uberhacker?

**Comment by uberhacker** [comment]
You can comment out this section in `crew` so it doesn't sync master.
```
 425   #update package lists                                                                                                                                                                  
 426   Dir.chdir CREW_LIB_PATH do                                                                                                                                                             
 427     system 'git fetch origin master'                                                                                                                                                     
 428     system 'git reset --hard origin/master'                                                                                                                                              
 429   end                                                                                                                                                                                    
 430   puts 'Package lists, crew, and library updated.'
```

**Comment by satmandu** [comment]
@uberhacker or I guess one could modify this from install.sh to point to their own git repo for checking to see if their modifications are working before submitting a PR?

```
cd "${CREW_LIB_PATH}"
rm -rf .git
git init
git remote add -f origin "https://github.com/${OWNER}/${REPO}.git"
git config core.sparsecheckout true
echo packages >> .git/info/sparse-checkout
echo lib >> .git/info/sparse-checkout
echo crew >> .git/info/sparse-checkout
git fetch origin "${BRANCH}"
git reset --hard origin/"${BRANCH}"
crew update
```

So that would be this, right?
```
# Set OWNER, REPO, and BRANCH env variables
cd /usr/local/lib/crew
git remote set-url origin https://github.com/${OWNER}/${REPO}.git
git fetch origin "${BRANCH}"
git reset --hard origin/"${BRANCH}"
crew update
```

**Comment by saltedcoffii** [comment]
I'm probably going to add a feature and command called `crew repo` where one can just specify the location of a default repo and add overlays as well. That way third parties can contribute to chromebrew, and we can have multiple repos like for testing or nonfree software. I'll do that after my `~/.bashrc.d` plan and writing up documentation.

**Pull request merged**